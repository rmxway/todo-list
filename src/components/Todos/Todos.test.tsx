import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Todos } from './';

const user = userEvent.setup();

describe('Todos', () => {
	it('Render is fine', () => {
		const { container } = render(<Todos />);

		expect(container).toMatchSnapshot();
	});

	it('Checked is fine', async () => {
		const { getByTestId } = render(<Todos />);
		const checkBoxFirstElement = getByTestId('1');

		await user.click(checkBoxFirstElement);

		expect(checkBoxFirstElement).toBeChecked();
	});

	it('Add new todo to normal', async () => {
		const { container, getByTestId } = render(<Todos />);
		const mainInput = getByTestId('main-input');
		const prevLength = container.querySelectorAll('[type="checkbox"]').length;

		await user.type(mainInput, 'New todo');
		await user.keyboard('[Enter]');

		const currentLength = container.querySelectorAll('[type="checkbox"]').length;

		expect(currentLength).toEqual(prevLength + 1);
	});
});
