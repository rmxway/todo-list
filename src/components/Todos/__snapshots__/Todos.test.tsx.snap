// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos Render is fine 1`] = `
<div>
  <div
    class="sc-aYaIB cqaHUp"
  >
    <i
      class="icon-angle-down"
    />
    <input
      data-testid="main-input"
      placeholder="What needs to be done?"
      type="text"
    />
    <label
      class="sc-gEvDqW hZyAII"
      for="1"
    >
      <i
        class="icon-check"
      />
      <input
        data-testid="1"
        id="1"
        name="1"
        role="checkbox"
        type="checkbox"
      />
      <span>
        Тестовое задание
      </span>
    </label>
    <label
      class="sc-gEvDqW hZyAII"
      for="2"
    >
      <i
        class="icon-check"
      />
      <input
        checked=""
        data-testid="2"
        id="2"
        name="2"
        role="checkbox"
        type="checkbox"
      />
      <span>
        Прекрасный код
      </span>
    </label>
    <label
      class="sc-gEvDqW hZyAII"
      for="3"
    >
      <i
        class="icon-check"
      />
      <input
        checked=""
        data-testid="3"
        id="3"
        name="3"
        role="checkbox"
        type="checkbox"
      />
      <span>
        Отличная компания
      </span>
    </label>
    <div
      class="sc-eqUzNf kCOKCG"
    >
      <div>
        1
         items left
      </div>
      <div>
        <label
          class="sc-dcJtft jzwPBj"
          for="all"
        >
          <input
            checked=""
            data-testid="filter-all"
            id="all"
            name="filter"
            type="radio"
          />
          All
        </label>
        <label
          class="sc-dcJtft jzwPBj"
          for="active"
        >
          <input
            data-testid="filter-active"
            id="active"
            name="filter"
            type="radio"
          />
          Active
        </label>
        <label
          class="sc-dcJtft jzwPBj"
          for="completed"
        >
          <input
            data-testid="filter-completed"
            id="completed"
            name="filter"
            type="radio"
          />
          Completed
        </label>
      </div>
      <div
        class="sc-fqkwJk bViZRM"
      >
        <i
          class="icon-refresh"
        />
        Clear completed
      </div>
    </div>
  </div>
</div>
`;
